src/query/functions/src/scalars/string.rs

```diff
@@ -239,6 +239,30 @@ pub fn register(registry: &mut FunctionRegistry) {
         }),
     );
 
+    registry.register_passthrough_nullable_3_arg::<StringType, StringType, StringType, StringType, _, _>(
+        "translate",
+        |_, _, _, _| FunctionDomain::Full,
+        vectorize_with_builder_3_arg::<StringType, StringType, StringType, StringType>(
+            |str, from, to, output, _| {
+            if from.is_empty() || from == to {
+                output.put_slice(str);
+                output.commit_row();
+                return;
+            }
+            let to_len = to.len();
```

Comment:
Should check: `from.len() == to.len()`?

Reply:
The length of from and to can be different

Replaces each character in string that matches a character in the from set with the corresponding character in the to set. If from is longer than to, occurrences of the extra characters in from are deleted.